src/main/java/com/bookshop/service/impl/BookServiceImpl.java

```diff
@@ -38,4 +38,17 @@ public BookDto findById(Long id) {
         return bookMapper.toDto(book);
 
     }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        findById(id);
+        Book book = bookMapper.toBook(requestDto);
+        book.setId(id);
+        return bookMapper.toDto(bookRepository.save(book));

```

Comment:
bad approach

you should fetch the desired book by id, then update it, and save afterwards

you can leverage mapstruct to perform update:
```
void updateBook(CreateBookRequestDto  dto, @MappingTarget Book book);
```

the final version will be like:
```
Book book = repo.findById(id).orElseThrow(...);
mapper.updateBook(dto, book);
return mapper.toDto(repo.save(book));
```